fdbclient/VersionVector.h

```diff
@@ -69,6 +73,46 @@ struct VersionVector {
 		maxVersion = invalidVersion;
 	}
 
+	void getDelta(Version version, VersionVector& delta) const {
+		ASSERT(version <= maxVersion);
+
+		delta.clear();
+
+		if (version == maxVersion) {
+			return; // rerurn an invalid version vector
+		}
+
+		std::map<Version, std::set<Tag>> versionMap;
+		for (auto& iter : versions) {
+			if (iter.second > version) {
+				versionMap[iter.second].insert(iter.first);
+			}
+		}
+
+		for (auto& iter : versionMap) {
+			delta.setVersion(iter.second, iter.first);
```

Comment:
I see `maxVersion` is not set in `setVersion(const std::set<Tag>& tags, Version version)`, which is correct for the logic here. But is that intended? Then the `delta` doesn't have `maxVersion` set.

Reply:
Good find, we need to set maxVersion in that method too. (I missed setting maxVersion in that method when I pulled it from Dan's PR, commit a107dd655e04ff69ea5369aefa8f825c062a4f99).

The assertion failures in simulation tests, which I haven't investigated yet, are probably a result of not setting maxVersion in this method.

Reply:
Note if you set `maxVersion` in that method, then next `setVersion` call would cause assertion failure.

Reply:
Sorry, why? Note we are ordering the entries (by version) before calling setVersion().

Reply:
You are right. I missed the part that `versionMap` is ordered.